src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -193,7 +193,7 @@ internal int VisualStudioVersion
 
         internal bool UseNewParser => ShouldUseNewParser(_solutionFile);
 
-        internal static bool ShouldUseNewParser(string solutionFile) => FileUtilities.IsSolutionXFilename(solutionFile);
+        internal static bool ShouldUseNewParser(string solutionFile) => FileUtilities.IsSolutionXFilename(solutionFile) || ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_14);
```

Comment:
nit: mostly this will reduce to a constant `true`, so it's more efficient to run this check _first_ and not have to do any string operations.

```suggestion
        internal static bool ShouldUseNewParser(string solutionFile) => ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_14) || FileUtilities.IsSolutionXFilename(solutionFile);
```

(here this is firmly in "who cares?" territory, that string check is cheap and we run it ~once per build. But I try to think about this sort of thing in all boolean-logic conditions; in a hot inner loop it can help.)

Reply:
makes sense. thanks!